developer/src/kmldmlc/src/keyman/kmx/kmx-builder.ts

```diff
@@ -195,20 +195,30 @@ export default class KMXBuilder {
       groupBase += KMXFile.COMP_GROUP_SIZE;
     }
 
-    size += this.calculateBitmapSize();
-    size += this.calculateKMXPlusSize();
+    size += this.buildBitmap();
+    size += this.buildKMXPlus(size);
 
     return size;
   }
 
-  calculateBitmapSize() {
+  buildBitmap() {
     // TODO
     return 0;
   }
 
-  calculateKMXPlusSize() {
-    // TODO
-    return 0;
+  buildKMXPlus(base: number) {
+    if(!(this.file instanceof KMXPlusFile)) {
+      return 0;
```

Comment:
should this be an error? or does 0 imply error?

Reply:
This is supposed to support non-kmxplus files as well, so this section would be zero-length, not an error, if not a kmxplus file.

Reply:
OK wasn't sure if this would be called in that case.